browser/resources/settings/brave_autofill_page/brave_autofill_page.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import {SettingsAutofillPageElement} from '../autofill_page/autofill_page.js'
+import {Router, RouteObserverMixin, Route} from '../router.js'
+
+const BaseElement = RouteObserverMixin(SettingsAutofillPageElement)
+
+export class BraveSettingsAutofillPageElement extends BaseElement {
+  static override get properties() {
+    const baseProperties = super.properties
+    const extended = Object.assign({}, baseProperties, {
+      isAutofillPage_: Boolean
+    })
+    return extended
+  }
+
+  private isAutofillPage_: boolean
+
+  override currentRouteChanged(newRoute: Route) {
+    this.isAutofillPage_ = newRoute == Router.getInstance().getRoutes().AUTOFILL
+  }
+}
```

Comment:
I think we can achieve this without the manual properties extending via Object.assign. Polymer automatically does this as far as I remember. I think this should work:

```ts
// Copyright (c) 2023 The Brave Authors. All rights reserved.
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this file,
// You can obtain one at https://mozilla.org/MPL/2.0/.

import {PolymerElement} from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js';
import {SettingsAutofillPageElement} from '../autofill_page/autofill_page.js'
import {Router, RouteObserverMixin, Route} from '../router.js'

type PolymerConstructor = new (...args: any[]) => PolymerElement;
const BaseElement: PolymerConstructor = SettingsAutofillPageElement

export class BraveSettingsAutofillPageElement extends RouteObserverMixin(BaseElement) {
  static get properties() {
    return {
      isAutofillPage_: Boolean
    }
  }

  isAutofillPage_: boolean

  override currentRouteChanged(newRoute: Route) {
    this.isAutofillPage_ = newRoute == Router.getInstance().getRoutes().AUTOFILL
  }
}
```

Reply:
Looks working, thanks, updated